src/types/ellipsis.rs

```diff
@@ -26,12 +26,22 @@ unsafe impl PyTypeInfo for PyEllipsis {
 
     #[inline]
     fn is_type_of(object: &PyAny) -> bool {
+        Self::is_type_of_bound(&object.as_borrowed())
+    }
+
+    #[inline]
+    fn is_type_of_bound(object: &Bound<'_, PyAny>) -> bool {
         // ellipsis is not usable as a base type
-        Self::is_exact_type_of(object)
+        Self::is_exact_type_of_bound(object)
     }
 
     #[inline]
     fn is_exact_type_of(object: &PyAny) -> bool {
+        Self::is_exact_type_of_bound(&object.as_borrowed())
+    }
+
+    #[inline]
+    fn is_exact_type_of_bound(object: &Bound<'_, PyAny>) -> bool {
         object.is(Self::get(object.py()))
```

Comment:
`PyEllipsis::get_bound` landed in the mean time, so maybe you can rebase and update here.